src/app/core/services/product.service.ts

```diff
@@ -42,4 +42,12 @@ export class ProductService {
    * @memberof ProductService
    */
   getProducts(): any { return this.http.get<Array<Product>>(`/spree/api/v1/products`) }
+
+  markAsFavorite(id: number): any { return this.http.post(`/spree/favorite_products`, {id: id}) }
```

Comment:
add return types for http methods here too `markAsFavorite(id: number): <`Return Type> { return this.http.post<Return Type here>(`/spree/favorite_products`, {id: id}) }`